ui/app/src/state/epics/configuration.ts

```diff
@@ -46,14 +55,37 @@ export default [
         )
       )
     ),
-  (action$, state$) =>
+  (action$, state$, { getIntl }) =>
     action$.pipe(
       ofType(fetchSiteConfig.type),
       withLatestFrom(state$),
       filter(([, state]) => Boolean(state.sites.active)),
       exhaustMap(([, state]) =>
         fetchSiteConfigService(state.sites.active).pipe(
-          map((config) => fetchSiteConfigComplete(config)),
+          switchMap((config) => {
+            try {
+              let localeCode = config.locale?.localeCode || state.uiConfig.locale.localeCode;
+              let options = config.locale?.dateTimeFormatOptions || state.uiConfig.locale.dateTimeFormatOptions;
+              new Intl.DateTimeFormat(localeCode, options);
+              return [fetchSiteConfigComplete(config)];
+            } catch (e) {
+              return [
+                fetchSiteConfigComplete({
+                  ...config,
+                  locale: {
+                    ...config.locale,
+                    localeCode: state.uiConfig.locale.localeCode,
+                    dateTimeFormatOptions: state.uiConfig.locale.dateTimeFormatOptions
+                  }
+                }),
+                showSystemNotification({
+                  // @ts-ignore - 'message' does not exist on type 'unknown', and catch variable type annotation must
+                  // be 'any' or 'unknown', but setting it to 'any' causes a prettier error.
+                  message: getIntl().formatMessage(configurationMessages.localeError, { message: e.message })
```

Comment:
```suggestion
                  message: getIntl().formatMessage(configurationMessages.localeError, { message: (e as Error).message })
```
Oddly, it works even without the casing for me. I didn't see the `'message' does not exist on type 'unknown'` TS error. Just removing the ts-ignore or even compiling. Do you get the error when compiling?

Reply:
I just realized that the error is being displayed only on the IDE, not when compiling. Should I leave it with the casting?